### PR TITLE
🎉 New source: Ip2whois [low-code SDK]

### PR DESCRIPTION
**Pull request by kaggrwal**
## What
https://www.ip2whois.com/developers-api [issue](https://github.com/airbytehq/connector-contest/issues/229)

## How
A new source connector using low-code cdk.

## Streams
- Whois

## 🚨 User Impact 🚨
No user impact

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [✓ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ✓] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ✓] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ✓] Code reviews completed
- [ ✓] Documentation updated
    - [ ✓] Connector's `README.md`
    - [ ✓] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [✓ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ✓] `docs/integrations/README.md`
    - [ ✓] `airbyte-integrations/builds.md`
- [ ✓] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

image

</details>

### CODE REVIEW

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Hello @kaggrwal, Marcos from Airbyte here 👋 . We received more than 25 new contributions along the weekend. One is yours 🎉  thank so much for! Our team is limited and maybe the review process can take longer than expected. As described in the Airbyte's Hacktoberfest your contribution was submitted before November 2nd and it is eligible to win the prize. The review process will validate other requirements. I ask to you patience until someone from the team review it. 

Because I reviewed some contributions for Hacktoberfest so far I saw some common patterns you can check in advance:
* [ ] Make sure you have added connector documentation to `/docs/integrations/`
* [ ] Remove the file `catalog` from `/integration_tests`
* [ ] Edit the `sample_config.json` inside `/integration_tests`
* [ ] For the `configured_catalog` you can use only `json_schema: {}`
* [ ] Add title to all properties in the `spec.yaml`
* [ ] Make sure the `documentationUrl` in the `spec.yaml` redirect to Airbyte's future connector page, eg: connector Airtable the `documentationUrl: https://docs.airbyte.com/integrations/sources/airtable`
* [ ] Review now new line at EOF (end-of-file) for all files.

If possible send to me a DM in Slack with the tests credentials, this process will make easier to us run integration tests and publish your connector. If you only has production keys, make sure to create a bootstrap.md explaining how to get the keys.

File: airbyte-integrations/connectors/source-ip2whois/source_ip2whois/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      airbyte_secret: true
+    domain:
+      title: Domain
+      type: string
+      description: >-
+        Domain name. See <a
+        href="https://www.ip2whois.com/developers-api">here</a>.
+      examples:
+        - www.google.com
+        - www.facebook.com
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-ip2whois/source_ip2whois/spec.yaml:25]: Why not use this as an array and use the ListStream feature to read records from multiple domains?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Small comment additional to the previous one.

**Comment by marcosmarxm** [comment]
@kaggrwal please check my first comment and check all tasks. Ask my review again when this is done.

**Review by marcosmarxm — APPROVED** [review]
Thanks @kaggrwal I'll publish later this week.

File: airbyte-integrations/connectors/source-ip2whois/source_ip2whois/schemas/TODO.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# TODO: Define your stream schemas
+Your connector must describe the schema of each stream it can output using [JSONSchema](https://json-schema.org). 
+
+You can describe the schema of your streams using one `.json` file per stream.
+ 
+## Static schemas
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-ip2whois/source_ip2whois/schemas/TODO.md:6]: please remove this file.

File: airbyte-integrations/connectors/source-ip2whois/source_ip2whois/schemas/whois.json [hunk 3]
```diff
@@ -0,0 +1,268 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "domain": {
+      "type": "string"
+    },
+    "domain_id": {
+      "type": "string"
+    },
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-ip2whois/source_ip2whois/schemas/whois.json:10]: The schema is not compatible:
```
The whois stream has the following schema errors:
'' is not a 'email'

Failed validating 'format' in schema['properties']['registrant']['properties']['email']:
    {'format': 'email', 'type': 'string'}

On instance['registrant']['email']:
    ''
--------------------------------------------------------------------------------
'' is not a 'email'

Failed validating 'format' in schema['properties']['admin']['properties']['email']:
    {'format': 'email', 'type': 'string'}

On instance['admin']['email']:
    ''
--------------------------------------------------------------------------------
'' is not a 'email'

Failed validating 'format' in schema['properties']['tech']['properties']['email']:
    {'format': 'email', 'type': 'string'}

On instance['tech']['email']:
    ''
```

File: airbyte-integrations/connectors/source-ip2whois/source_ip2whois/spec.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+documentationUrl: https://docs.airbyte.com/integrations/sources/ip2whois
+connectionSpecification:
+  $schema: http://json-schema.org/draft-07/schema#
+  title: Ip2whois Spec
+  type: object
+  required:
+    - access_key
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-ip2whois/source_ip2whois/spec.yaml:7]: `access_key` doesn't exist.

File: docs/integrations/README.md [hunk 5]
```diff
@@ -87,7 +87,8 @@ For more information about the grading system, see [Product Release Stages](http
 | [HubSpot](sources/hubspot.md)                                                               | Generally Available  | Yes                 |
 | [Insightly](sources/insightly.md)                                                           | Alpha                | Yes                 |
 | [Instagram](sources/instagram.md)                                                           | Generally Available  | Yes                 |
-| [Intercom](sources/intercom.md)                                                             | Generally Available  | Yes                 |
+| [Intercom](sources/intercom.md)                                                             | Generally Available                  Yes 
+| [Ip2whois](source/ip2whois.md)                                                              | Alpha                | Yes                 |
```
Comment (marcosmarxm) [inline docs/integrations/README.md:91]:
```suggestion
| [Ip2whois](source/ip2whois.md)                                                              | Alpha                | No                 |
```

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
I found some problems running tests, please fix them.

**Comment by marcosmarxm** [comment]
Hello! I'm going to be out of the office this Friday and won't be able to review your contribution again today, I return next Monday. So far, most contributions look solid and are almost done to be approved. As said in [Chris' comment](https://airbytehq.slack.com/archives/C0450PCSDB2/p1667430678866629) all contributions made before 2-November are eligible to receive the prize and have 2 weeks to merge the contributions. But I ensure next week we're going to have your contribution merged. If you have questions about the implementation you can send them in `#hacktoberfest-2022` in Airbyte's Slack.

Sorry the inconvenience and see you again next week, thank you so much for your contribution!

**Comment by marcosmarxm** [comment]
@kaggrwal let me know when you solve the issues.

**Comment by kaggrwal** [comment]
> @kaggrwal let me know when you solve the issues.

@marcosmarxm it's working fine i have tested it, to get API keys, you need to create an account https://www.ip2location.io/ or i can share my keys for the testing purposes.

**Review by marcosmarxm — APPROVED** [review]
Thanks @kaggrwal

**Pull request merged**